### DISCUSSION TITLE
Is it possible to create geometry like a pyramid?

### DISCUSSION THREAD

**Discussion by leSemaleon** [NanoComp/meep#2721] [Q&A]
I did not find such geometry on the site.  The closest thing is a prism and a cone

**Comment by stevengj** [comment]
You will have to build it out of other geometric objects (or use a `material-function`).  e.g. you could divide the pyramid into blocks in various ways (e.g. starting with a solid block and "chopping off" pieces with blocks of air at various angles).

Meep doesn't currently have a "generic polyhedron" object.   (In principle we could implement support for e.g. any closed surface described by a triangular mesh, e.g. generated with gmsh.)

See also https://github.com/NanoComp/meep/discussions/2627 and https://github.com/NanoComp/libctl/issues/67
